public/app/plugins/datasource/elasticsearch/datasource.ts

```diff
@@ -892,6 +895,17 @@ export class ElasticDatasource
     return false;
   }
 
+  analyzeQuery(query: ElasticsearchQuery, options: AnalyzeQueryOptions): boolean {
+    let expression = query.query ?? '';
+    switch (options.check) {
+      case 'HAS_FILTER': {
+        return queryHasFilter(expression, options.attributes.key, options.attributes.value);
+      }
+      default:
+        return false;
+    }
+  }
+
   modifyQuery(query: ElasticsearchQuery, action: QueryFixAction): ElasticsearchQuery {
```

Comment:
Can we here also keep original `modifyQuery` and add a new method for toggling label filters?